aea/mail/base.py

```diff
@@ -209,7 +209,7 @@ def encode(self, envelope: "Envelope") -> bytes:
         envelope_pb.sender = envelope.sender
         envelope_pb.protocol_id = str(envelope.protocol_id)
         envelope_pb.message = envelope.message_bytes
-        if envelope.context is not None:
+        if envelope.context.uri_raw != "":
```

Comment:
Context is optional; so I think this check is incorrect

Reply:
it is instantiated by default if not provided to `__init__`. Also, the `context` property is not optional.